R/Filtering.R

```diff
@@ -502,3 +502,251 @@ FilterParFieldReview <- function(field, .data, parameter) {
   print(pie)
   print(df)
 }
+
+#' Unique Field Values Table
+#'
+#' Function creates a table yielding the number of unique values in each field (column) returned.
+#' The first column of the summary table holds the field name and the second column holds the
+#' unique value count.
+#' 
+#' @param .data TADA dataframe
+#' @param display A character string denoting what fields to return in the summary table. Defaults to "key". "all" will return all fields in the dataset, "narrow" will return most field names except those holding numeric values or units, and "key" returns the most important columns to review. Note that if a field is completely NA, it will not be shown on the summary table.
+#' @param characteristicName Optional. Defaults to "null". A vector of TADA-converted (all caps) WQP characteristics a user may provide to filter the results to one or more characteristics of interest. "null" will show a summary table for the whole dataset.
+#'
+#' @return A summary table yielding the number of unique values in each field.
+#'
+#' @export
+#' 
+#' @examples 
+#' # Load example dataset:
+#' data(Nutrients_Utah)
+#' # Count table of key fields in Nutrients_Utah dataset
+#' fieldCountUT <- fieldCounts(Nutrients_Utah)
+#' # Count table of most fields in Nutrients_Utah, filtered to only AMMONIA results.
+#' fieldCountUTAmmonia <- fieldCounts(Nutrients_Utah, display = "narrow", characteristicName = "AMMONIA")
+
+fieldCounts <- function(.data, display = c("key","narrow", "all"), characteristicName = "null"){
+  # check .data is data.frame
+  checkType(.data, "data.frame", "Input object")
+  
+  display = match.arg(display)
+  
+  # filter to characteristic if provided
+  if(!characteristicName%in%c("null")){
+    .data = subset(.data, .data$TADA.CharacteristicName%in%c(characteristicName))
+    
+    if(dim(.data)[1]<1){
+      stop("Characteristic name(s) provided are not contained within the input dataset. Note that TADA converts characteristic names to ALL CAPS for easier harmonization.")
```

Comment:
... easier harmonization of synonyms from different WQP data providers (USGS's NWIS and EPA's WQX).

Reply:
> ... easier harmonization of synonyms from different WQP data providers (USGS's NWIS and EPA's WQX).

**Resolved**